client/hotrod-client/src/main/java/org/infinispan/client/hotrod/DataFormat.java

```diff
@@ -24,44 +25,161 @@ public final class DataFormat {
 
    private static final Log log = LogFactory.getLog(DataFormat.class, Log.class);
 
-   private final MediaType keyType;
-   private final MediaType valueType;
-   private final Marshaller keyMarshaller;
-   private final Marshaller valueMarshaller;
+   private final BufferSizePredictor keySizePredictor = new AdaptiveBufferSizePredictor();
+   private final BufferSizePredictor valueSizePredictor = new AdaptiveBufferSizePredictor();
 
+   private final DataFormatImpl server;
+   private final DataFormatImpl client;
    private MarshallerRegistry marshallerRegistry;
    private Marshaller defaultMarshaller;
    private boolean isObjectStorage;
```

Comment:
Removed the boolean.

Yeah, I wanted to remove most methods and use only the `client()` and `server()` methods. But I am afraid this would be a breaking change to backport.